deps/v8/src/common/globals.h

```diff
@@ -404,6 +404,7 @@ enum TypeofMode : int { INSIDE_TYPEOF, NOT_INSIDE_TYPEOF };
 // Enums used by CEntry.
 enum SaveFPRegsMode { kDontSaveFPRegs, kSaveFPRegs };
 enum ArgvMode { kArgvOnStack, kArgvInRegister };
+enum FunctionDescriptorMode { kNoFunctionDescriptor, kHasFunctionDescriptor };
```

Comment:
Shouldn't this have been `enum FunctionDescriptorMode : bool { /* ... */ }`, or (probably more idiomatic for V8) use the enum instead of `bool`? Relying on int-to-bool coercion looks a bit footgunny.

Reply:
I'm guessing they were ok with it as it's following the previous enums `SaveFPRegsMode ` and `ArgvMode`.

Reply:
Right, but those are of type `SaveFPRegsMode` and `ArgvMode` in function parameter lists, there's no coercion to bool taking place.

Reply:
that's right however the int value `0` (kNoFunctionDescriptor) is implicitly translating to `false` at all cases, and true otherwise. You're right that marking it as `bool` could have made it more clear.